### PR TITLE
Sdss 91 error output

### PR DESCRIPTION
**Pull request by IsaiahLai111**

### CODE REVIEW

File: src/app/exception_throw.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
+from fastapi import status
+from fastapi.responses import JSONResponse
+
+
+class exception_response:
```
Comment (JackRimmerAND) [inline src/app/exception_throw.py:5]: You should be following class naming conventions i.e. Pascal Case, so this should be ExceptionResponse. Also I don't think this is the best name for the class either, I'd call it maybe ExceptionResponder. Also the class name and the file name should be consistent, when importing this class you should probably be doing e.g.

from exception_responder import ExceptionResponder

File: src/app/app.py [hunk 2]
```diff
@@ -10,6 +12,25 @@
 app = FastAPI()
 
 
+@app.exception_handler(500)
+async def internal_exception_handler(request: Request, exc: Exception):
+    """
```
Comment (JackRimmerAND) [inline src/app/app.py:17]: How come you're passing 'request' and 'exc' as parameters when they aren't being used in here?

File: src/app/app.py [hunk 3]
```diff
@@ -10,6 +12,25 @@
...
+    Override the global exception handler (500 internal server error) in
+    FastAPI and throw error in JSON format
+    """
+    return exception_throw.throw_500_global_exception()
+
+
+@app.exception_handler(RequestValidationError)
+async def validation_exception_handler(request: Request, exc: RequestValidationError):
```
Comment (JackRimmerAND) [inline src/app/app.py:25]: Same issue as above

File: src/app/app.py [hunk 4]
```diff
@@ -61,10 +82,16 @@ async def get_schema(survey_id: str, version: str) -> dict:
     logger.info("Getting schema metadata...")
     logger.debug(f"Input data: survey_id={survey_id}, version={version}")
 
+    try:
```
Comment (JackRimmerAND) [inline src/app/app.py:85]: I wouldn't have this validation logic in app.py, ideally we'd abstract validation logic into a separate Validator class

File: src/app/app.py [hunk 5]
```diff
@@ -73,19 +100,30 @@ async def get_schema(survey_id: str, version: str) -> dict:
...
+
     logger.info("Schema successfully retrieved.")
     logger.debug(f"Schema: {schema}")
 
     return schema
 
 
 @app.get("/v1/schema_metadata", response_model=list[ReturnedSchemaMetadata])
-async def get_schemas_metadata(survey_id: str) -> list[ReturnedSchemaMetadata]:
+async def get_schemas_metadata(survey_id: str = "") -> list[ReturnedSchemaMetadata]:
```
Comment (JackRimmerAND) [inline src/app/app.py:114]: How come we're setting a default value here? What happens if we remove this then on line 116 check 
`if survey_id is None`
instead?

File: src/app/app.py [hunk 6]
```diff
@@ -95,12 +133,16 @@ async def get_schemas_metadata(survey_id: str) -> list[ReturnedSchemaMetadata]:
 
 @app.get("/v1/dataset_metadata", response_model=list[DatasetMetadata])
 async def get_dataset_metadata_collection(
-    survey_id: str, period_id: str
+    survey_id: str = "", period_id: str = ""
```
Comment (JackRimmerAND) [inline src/app/app.py:136]: Same issue as above, how come we're setting default values here?

File: src/app/database.py [hunk 7]
```diff
@@ -122,6 +122,8 @@ def get_schema_metadata(survey_id, version) -> SchemaMetadata:
 
     for schema in schemas_result:
         return_metadata = schema.to_dict()
+        if len(return_metadata) == 0:
+            return None
```
Comment (JackRimmerAND) [inline src/app/database.py:126]: I don't like the idea of returning `None` here, this should probably pass the issue straight to the exception filter

File: src/app/exception_throw.py [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+        self.message = message
+
+    def throw_er_with_json(self) -> JSONResponse:
+        return JSONResponse(
+            status_code=self.status_code,
+            content={"status": self.status, "message": self.message},
+        )
+
+
+def throw_500_global_exception() -> JSONResponse:
```
Comment (JackRimmerAND) [inline src/app/exception_throw.py:22]: How come you've defined an exception_response class and separate interceptor functions in the same file? I would separate these two concepts into different files, and probably wrap these throw functions inside an ExceptionInterceptor class.

**Review by JackRimmerAND — CHANGES_REQUESTED** [review]

File: src/app/app.py (hunk 6 above)
Comment (IsaiahLai111) [reply src/app/app.py]: Changed empty string to None

File: src/app/database.py (hunk 7 above)
Comment (IsaiahLai111) [reply src/app/database.py]: Removed this logic

File: src/app/exception_throw.py (hunk 8 above)
Comment (IsaiahLai111) [reply src/app/exception_throw.py]: Refactored into two files

File: src/app/storage.py [hunk 9]
```diff
@@ -29,5 +29,8 @@ def store_schema(schema: Schema, schema_id):
 
 def get_schema(filename):
     """Get the SDS schema from the schema bucket using the filename provided."""
-    schema = json.loads(bucket.blob(filename).download_as_string())
-    return schema
+    try:
+        schema = json.loads(bucket.blob(filename).download_as_string())
+        return schema
+    except Exception:
+        return None
```
[the remark this answers is not in the document]
Comment (IsaiahLai111) [reply src/app/storage.py]: Raised a custom exception at this point

File: src/app/app.py (hunk 5 above)
Comment (IsaiahLai111) [reply src/app/app.py]: Changed from empty string to None

File: src/app/app.py (hunk 4 above)
Comment (IsaiahLai111) [reply src/app/app.py]: Moved logic to a separate file and class

File: src/app/app.py (hunk 3 above)
Comment (IsaiahLai111) [reply src/app/app.py:52]: Requirement by FASTapi on error handler

File: src/app/exception_throw.py (hunk 1 above)
Comment (IsaiahLai111) [reply src/app/exception_throw.py]: Changed the name to follow the naming conventions

**Review by JackRimmerAND — APPROVED** [review]

**Review by AND-JAshcroft — APPROVED** [review]

**Pull request merged**